lib/Attachment.php

```diff
@@ -156,6 +156,53 @@ public function __toString() {
 	 * @param int|mixed $iid An attachment identifier.
 	 */
 	public function init( $iid = null ) {
+		$iid = $this->determine_id( $iid );
+
+		/**
+		 * determine_id returns null when the attachment is a file path,
+		 * thus there's nothing in the DB for us to do here 
+		 */
+		if ( null === $iid ) {
+			return null;
```

Comment:
The `init()` function probably shouldn’t return anything, right? I think we should just `return` here. Then we can also remove the `return null;` statement at the end of this function.

Reply:
What about returning null on failure and itself on creation. One thing I currently don't like about `Image()` is that it will return an object even though the constructor argument might be entirely false. Think about `Image(99999)`. This is not correct but will not cancel or throw an error. I think the `Image()` function should return `null` if the initialisation fails and hence Twig will render nothing on output. What do you think?

Reply:
Agreed. Being "forgiving" with `null`s was a bad move, let's fix it